cmd/captplanet/run.go

```diff
@@ -145,9 +147,10 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 		errch <- runCfg.Uplink.Run(ctx)
 	}()
 
-	for v := range errch {
-		err = fmt.Errorf("%s : %s", err, v)
+	select {
+	case error := <-errch:
```

Comment:
this is not reading all the errors

Reply:
What do you mean? It's only going to pick up errors for this error channel, that is correct, but I thought that was the scope of this card. If it needs to pick up different errors, I can reformat it to do so

Reply:
Let's say 3 of the storage nodes fail then it should get three errors as a result, not just one.

Reply:
The card I was assigned specifically wanted me to use the utils.CombineErrors function, though. Are you saying we shouldn't use CombineErrors here?

Reply:
```
for v := range errch {
   errs = append(errs, err)
}

return utils.CombineErrors(errs...)
```

However I'm not quite sure whether that's sufficient since nothing is closing the channel at the moment. Waiting for all the errors would be also difficult.

You could actually just use:
```
return <-errch
```

But, yeah, getting multiple errors would be better. One possibility would be to write something that gets the first error and then waits few more seconds for errors and finally combines them together and returns that. This would give more information and would be easier to implement.